### PR TITLE
Fix onchainkit

### PR DESCRIPTION
**Pull request by rin-st**
Up versions of deps 

onchainkit inner deps is
```
    "@tanstack/react-query": "^5",
    "viem": "^2.17.4",
    "wagmi": "^2.11.0"
```
so looks like nothing should break if they won't change major versions. Previously it could break because of intersection with `create-eth` versions of that deps, or because of fixed versions.

+ fixed one import after changing onchainkit version

---

To test

```
npx create-eth@latest -e scaffold-eth/create-eth-extensions:fix-onchainkit
```

---

note: asked Austin to add avatar to base ens, currently it looks like this
photo_2024-10-02 19 14 39

### CODE REVIEW

**Comment by technophile-04** [comment]
> so looks like nothing should break if they won't change major versions. Previously it could break because of intersection with create-eth versions of that deps, or because of fixed versions.

Didn't get this 😅, it should still break right in near future like if any of the dependency changes their version (wagmi, viem, tanstack)? 

Also @rin-st if we remove this file : 

https://github.com/scaffold-eth/create-eth-extensions/blob/onchainkit/extension/package.json

it breaks again. And because of that file to this dependency in this PR does is not actually taking place: 

This PR deps: 
```
"@tanstack/react-query": "^5.59.0",
"viem": "^2.21.6",
"wagmi": "^2.12.16"
```

Actually if you do `yarn why  @tanstack/react-query` it should only show `5.29.0` (becuase of resolution at root package.json)

**Comment by rin-st** [comment]
> Didn't get this 😅, it should still break right in near future like if any of the dependency changes their version (wagmi, viem, tanstack)?

As I understand dependencies can break in situations:
1. We fix our versions of wagmi/react-query/viem. For example wagmi: 1.0.0. Onchainkit inner deps cannot be fixed without resolutions. So it could be for example wagmi: ^1.0.0. and it takes last version when installed, for example 1.2.3. So versions of providers(our version)/consumers(their version)/something else related to wagmi/react-query/viem could be incompatible.
2. Sometimes if you use right version in extension, but `create-eth` version is fixed and that versions has intersection, `merge-packages` takes that intersection and we're returning to point 1. For example, version inside extension "wagmi": "^1.0.0". Version inside create-eth "wagmi": "1.1.0" or "~1.1.0". Intersection is "1.1.0" or "~1.1.0". So resulting version of extension will be that intersection - "1.1.0" or "~1.1.0".

Current situation regarding onchainkit. No fixed versions (1), no intersections (2), major versions of deps of onchainkit lib and onchainkit extensions are the same https://github.com/scaffold-eth/create-eth-extensions/pull/30#issue-2562194473, so as I understand nothing should break on install

> Also @rin-st if we remove this file :
https://github.com/scaffold-eth/create-eth-extensions/blob/onchainkit/extension/package.json
it breaks again.

I removed it and it works fine for me. Could you check again?

> And because of that file to this dependency in this PR does is not actually taking place:

I don't understand this)

> Actually if you do yarn why  @tanstack/react-query it should only show 5.29.0 (becuase of resolution at root package.json)

```
yarn why @tanstack/react-query
├─ @coinbase/onchainkit@npm:0.33.3
│  └─ @tanstack/react-query@npm:5.59.0 (via npm:^5)
│
├─ @coinbase/onchainkit@npm:0.33.3 [ef549]
│  └─ @tanstack/react-query@npm:5.59.0 [ef549] (via npm:^5.59.0 [ef549])
│
└─ @se-2/nextjs@workspace:packages/nextjs
   └─ @tanstack/react-query@npm:5.59.0 [ef549] (via npm:^5.59.0 [ef549])
```

**Comment by technophile-04** [comment]
> Current situation regarding onchainkit. No fixed versions (1), no intersections (2), major versions of deps of onchainkit lib and onchainkit extensions are the same https://github.com/scaffold-eth/create-eth-extensions/pull/30#issue-2562194473, so as I understand nothing should break on install

Kind of make sense! But I don't get it why it started breaking earlier? Since we had similar dependency versions previously right to make it work? 

pic of early version of dependencies when it was breaking: 
Screenshot 2024-10-03 at 11 20 47 AM

> I removed it and it works fine for me.

Ohh just noticed it works with `npx create-eth@latest` but when we release the new version (i.e merge https://github.com/scaffold-eth/create-eth/pull/126 it breaks with that) 

To reproduce: 

1. Make sure you have `create-eth` repo clonned and cd into `create-eth`.
2. Switch to [`backmerge-2-10-24-new`](https://github.com/scaffold-eth/create-eth/pull/126) branch ( `git switch backmerge-2-10-24-new` ) 
3. Run the dev server : `yarn dev`
4. Then `cd externalExtensions` and clone this `create-eth-extensions` inside it. 
5. then cd into `create-eth-extensions` the actual path should look like `create-eth/externalExtensions/create-eth-extensions`
6. Switch to this branch `git switch fix-onchainkit`
7. Go to the root of `create-eth` and run `yarn cli -e create-eth-extensions --dev`  <= this will give you the changes of backmerge-2-10-24-new + `fix-onchainkit` branch. 

Now if you navigate to `localhost:3000/onchainkit-example`

<details>

<summary>
I see this : 
</summary>

image


</details>

> I don't understand this)

Ohh I was saying this when we forgot to remove the root package.json resolutions, since you already removed it now all good! and now `yarn why` also shows me tansactack 5.59.0 version.

**Comment by rin-st** [comment]
> But I don't get it why it started breaking earlier? Since we had similar dependency versions previously right to make it work?

> Ohh just noticed it works with npx create-eth@latest but when we release the new version (i.e merge https://github.com/scaffold-eth/create-eth/pull/126 it will break with that)

Basically, it's point 2 from here https://github.com/scaffold-eth/create-eth-extensions/pull/30#issuecomment-2389491891 in both cases.

For some reason I used not the last version of viem in this pr. It was "^2.21.6". Version of viem in `[backmerge-2-10-24-new](https://github.com/scaffold-eth/create-eth/tree/backmerge-2-10-24-new)` of `create-eth` is "2.21.7". Intersection of that versions is "2.21.7", so when creating instance with onchainkit **extension** resulting version of "viem" is "2.21.7". But onchainkit **lib** uses last viem version "2.21.17" as internal dependency. So wagmi inside `create-eth` uses viem "2.21.7" and creates wagmi provider, and `wagmi` inside `onchainkit` **lib** uses viem "2.21.17" and relies on outer provider with different version of viem and that causes bugs. Not sure about the last sentence but I think 90% I'm right.

I hope I made it a bit clearer

Now I updated viem to "^2.21.17" and error is gone.

---

To sum up: What can break onchainkit **extension** again? If version of one of react-query, wagmi or viem inside `create-eth` will be fixed from the top "x.y.z" or "~x.y.z" or "<x.y.z" **and at the same time** one of the react-query, wagmi or viem version inside `onchainkit` **extension** will be <= version of that lib in `create-eth`.

So we need to keep versions inside `onchainkit` **extension** > than versions inside `create-eth` if versions of libs inside `create-eth` are fixed. It could be equal if it's very last version, but it will break after some updates so it's not stable.

If versions of libs inside `create-eth` would be not fixed - "^x.y.z", we'd just need to keep equal major versions inside `onchainkit` **extension** and `create-eth`

**Review by technophile-04 — APPROVED** [review]

**Comment by rin-st** [comment]
> Lol to me I don't see any difference b/w dependencies versions not sure what caused earlier version to break.

See viem @se-2/nextjs versions. On first screenshot it is 2.21.7 it breaks because of it. On second it is 2.21.17 so everything is ok in that case.

I tried to explain it here

> For some reason I used not the last version of viem in this pr. It was "^2.21.6". Version of viem in backmerge-2-10-24-new branch of create-eth is "2.21.7". Intersection of that versions is "2.21.7", so when creating instance with onchainkit extension resulting version of "viem" is "2.21.7". But onchainkit lib uses last viem version "2.21.17" as internal dependency. So wagmi inside create-eth uses viem "2.21.7" and creates wagmi provider, and wagmi inside onchainkit lib uses viem "2.21.17" and relies on outer provider with different version of viem and that causes bugs. Not sure about the last sentence but I think 90% I'm right.

**Comment by technophile-04** [comment]
Ohhhhhhhhhh damn man sorry, I was overlooking the difference b/w `.17` and `.7` and was thinking even though they both have same version why is breaking!!

> I tried to explain it here

Now this makes complete sense! Tysm!!!
